source/fem-dem/ib_particles_dem.cc

```diff
@@ -67,6 +67,69 @@ IBParticlesDEM<dim>::update_particles(
   cfd_time      = time;
 }
 
+template <int dim>
+void
+IBParticlesDEM<dim>::update_contact_candidates()
+{
+  particles_contact_candidates.resize(dem_particles.size());
+
+  double radius_factor = parameters->contact_search_radius_factor;
+
+  // If contact_search_radius_factor < 1, all particles are taken into
+  // account in the contact search. Otherwise, do it for the particles inside
+  // the region of contact search. By default, radius_factor is zero, so all
+  // particles are taken into account.
+  if (radius_factor <= 1)
```

Comment:
I don't like these weird behaviors based on a double. If you want all particles to take into account all particles, but a very large value of the radius factor. There is no reason why <0 would lead to all particles taking into account all particles. It's like changing the meaning of the parameter.